plugins/net_plugin/net_plugin.cpp

```diff
@@ -3082,7 +3185,29 @@ namespace eosio {
          }
       }
 
-      my_impl->sync_master->recv_handshake( shared_from_this(), msg );
+      uint32_t nblk_combined_latency = update_latency(msg);
+      my_impl->sync_master->recv_handshake( shared_from_this(), msg, nblk_combined_latency );
+   }
+
+   // called from connection strand
+   uint32_t connection::update_latency(const handshake_message& msg) {
```

Comment:
Instead of using a latency based on time difference between the two peers (which could be very wrong depending on how well both peers synchromize their clocks), maybe it woul be better to base it on how fast a round trip takes (like for the handshake).
Besides not relying on the fact that the two peers clocks are synchronized, it might also take into account how busy (responsive) the node is.

Reply:
I like this idea. I plan on having one additional PR for this issue. How about that comes in with the next one?

Reply:
Sure, let's get this PR in first!